httpclient-backend/zio/src/main/scala/sttp/client3/httpclient/zio/ZioBodyFromHttpClient.scala

```diff
@@ -1,77 +1,112 @@
+                )
+                bytes <- readAllBytes(fileChannel)
+              } yield bytes
+            )
         }
       }
 
-      override protected def regularIgnore(response: ZStream[Blocking, Throwable, Byte]): BlockingTask[Unit] =
+      override protected def regularIgnore(response: ZStream[Any, Throwable, Byte]): Task[Unit] =
         response.run(ZSink.drain)
 
       override protected def regularAsByteArray(
-          response: ZStream[Blocking, Throwable, Byte]
-      ): BlockingTask[Array[Byte]] = response.runCollect.map(_.toArray)
+          response: ZStream[Any, Throwable, Byte]
+      ): Task[Array[Byte]] = response.runCollect.map(_.toArray)
 
       override protected def regularAsFile(
-          response: ZStream[Blocking, Throwable, Byte],
+          response: ZStream[Any, Throwable, Byte],
           file: SttpFile
-      ): BlockingTask[SttpFile] = response.run(ZSink.fromFile(file.toPath)).as(file)
+      ): Task[SttpFile] = response
+        .run({
+          ZSink.managed(
+            AsynchronousFileChannel
+              .open(Path.fromJava(file.toPath), StandardOpenOption.WRITE, StandardOpenOption.CREATE)
+          ) { fileChannel =>
+            ZSink.foldChunksM(0L)(_ => true) { case (position, data) =>
+              fileChannel
+                .write(data, position)
+                .map(bytesWritten => position + bytesWritten)
+            }
+          }
+        })
+        .as(file)
 
       override protected def regularAsStream(
-          response: ZStream[Blocking, Throwable, Byte]
-      ): Task[(ZStream[Blocking, Throwable, Byte], () => BlockingTask[Unit])] =
+          response: ZStream[Any, Throwable, Byte]
+      ): Task[(ZStream[Any, Throwable, Byte], () => Task[Unit])] =
         Task.succeed((response, () => response.runDrain.catchAll(_ => ZIO.unit)))
 
       override protected def handleWS[T](
           responseAs: WebSocketResponseAs[T, _],
           meta: ResponseMetadata,
-          ws: WebSocket[BlockingTask]
-      ): BlockingTask[T] = bodyFromWs(responseAs, ws)
+          ws: WebSocket[Task]
+      ): Task[T] = bodyFromWs(responseAs, ws)
 
       override protected def cleanupWhenNotAWebSocket(
-          response: ZStream[Blocking, Throwable, Byte],
+          response: ZStream[Any, Throwable, Byte],
           e: NotAWebSocketException
-      ): BlockingTask[Unit] = response.run(ZSink.drain)
+      ): Task[Unit] = response.run(ZSink.drain)
 
       override protected def cleanupWhenGotWebSocket(
-          response: WebSocket[BlockingTask],
+          response: WebSocket[Task],
           e: GotAWebSocketException
-      ): BlockingTask[Unit] = response.close()
+      ): Task[Unit] = response.close()
     }
 
-  override implicit def monad: MonadError[BlockingTask] = new RIOMonadAsyncError[Blocking]
+  private def readAllBytes(fileChannel: AsynchronousFileChannel) = {
+    val bufferSize = 4096
```

Comment:
Should we restore chunkSize parameter in constructor and pass this value all the way down to here?

Reply:
I can't imagine really a situation where you'd want to customise this. Let's leave it, 4K seems like a good magic number ;)